core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/LimitTest.java

```diff
@@ -45,7 +45,7 @@ public void builderShouldCreatePopulatedOperation() {
         assertThat(limit.getInput())
                 .hasSize(2);
         assertThat(limit.getResultLimit()).isEqualTo(1);
-        assertThat(limit.getInput()).containsOnly("1", "2");
+        assertThat((Iterable<String>) limit.getInput()).containsOnly("1", "2");
```

Comment:
Could use the following instead to avoid casting:
`assertThat(limit.getInput())..asInstanceOf(InstanceOfAssertFactories.iterable(String.class)).containsOnly("1", "2");`
Or:
`assertThat(limit.getInput())..asInstanceOf(InstanceOfAssertFactories.list(String.class)).containsOnly("1", "2");`

Reply:
That is another way to do it, but it's longer and there's still a cast, it's just hidden inside [InstanceOfAssertFactory](https://javadoc.io/static/org.assertj/assertj-core/3.22.0/org/assertj/core/api/InstanceOfAssertFactory.html) instead of being in this class. With the way generics are used there's no way to escape the cast.

Reply:
The shorter version is what I suggested: https://github.com/gchq/Gaffer/pull/2624#discussion_r843135764
But I still agree this should be fixed everywhere in a different ticket.

Reply:
This can be cleaned up later under https://github.com/gchq/Gaffer/issues/2625